src/mode/java_highlight_rules.js

```diff
@@ -145,6 +121,143 @@ var JavaHighlightRules = function() {
             }, {
                 defaultToken : "comment"
             }
+        ],
+        "strings": [
+            {
+                token: "string",
+                regex: /(?:STR|FMT)\."/,
```

Comment:
This isn't correct. Technically the prefix can be any expression so long as it returns a TemplateProcessor

so

```java
try (var conn = ds.getConnection()) {
   var stmt = conn."""
   SELECT * FROM table WHERE x = \{x}
   """;
}
```

or 

```java
a.b().c().e()."""
   Sub \{here}
"""

Reply:
In that case, new approach should work. The only problem I see - is usage of outstanding ."""text""" without adding any processor, but for solving this case mode should recognize whole contexts and this would be a little bit too much.